src/main/webapp/app/exercises/programming/manage/programming-exercise-edit-selected.component.ts

```diff
@@ -0,0 +1,113 @@
+        if (!this.newProgrammingExercise.releaseDate) {
+            const confirmNoReleaseDate = this.translateService.instant(this.translationBasePath + 'noReleaseDateWarning');
+            if (!window.confirm(confirmNoReleaseDate)) {
+                return;
+            }
+        }
+        this.isSaving = true;
+        this.selectedProgrammingExercises.forEach((programmingExercise) => {
+            programmingExercise = this.setNewValues(programmingExercise);
+            if (programmingExercise.id !== undefined) {
+                const requestOptions = {} as any;
+                if (this.notificationText) {
+                    requestOptions.notificationText = this.notificationText;
+                }
+                this.subscribeToSaveResponse(programmingExercise.title, this.programmingExerciseService.update(programmingExercise, requestOptions));
+            } else if (programmingExercise.noVersionControlAndContinuousIntegrationAvailable) {
+                // only for testing purposes(noVersionControlAndContinuousIntegrationAvailable)
+                this.subscribeToSaveResponse(programmingExercise.title, this.programmingExerciseSimulationService.automaticSetupWithoutConnectionToVCSandCI(programmingExercise));
+            } else {
+                this.subscribeToSaveResponse(programmingExercise.title, this.programmingExerciseService.automaticSetup(programmingExercise));
+            }
+        });
+    }
+
+    /**
+     * Replace the programming exercise values with the new given ones
+     * @param programmingExercise to update
+     * @return the programming exercise with updated values
+     */
+    setNewValues(programmingExercise: ProgrammingExercise) {
+        programmingExercise.releaseDate = this.newProgrammingExercise.releaseDate;
+        programmingExercise.dueDate = this.newProgrammingExercise.dueDate;
+        programmingExercise.buildAndTestStudentSubmissionsAfterDueDate = this.newProgrammingExercise.buildAndTestStudentSubmissionsAfterDueDate;
+        programmingExercise.assessmentDueDate = this.newProgrammingExercise.assessmentDueDate;
+        return programmingExercise;
+    }
+
+    private subscribeToSaveResponse(exerciseTitle: string | undefined, result: Observable<HttpResponse<ProgrammingExercise>>) {
+        result.subscribe(
+            () => this.onSaveSuccess(),
+            (res: HttpErrorResponse) => this.onSaveError(res, exerciseTitle),
+        );
+    }
+
+    private onSaveSuccess() {
+        this.savedExercises++;
+        if (this.savedExercises === this.selectedProgrammingExercises.length) {
+            this.isSaving = false;
+            if (this.failedExercises.length <= 0) {
+                this.activeModal.close();
+            }
+        }
+    }
+
+    private onSaveError(error: HttpErrorResponse, exerciseTitle?: string | undefined) {
+        exerciseTitle = exerciseTitle == undefined ? 'undefined exercise' : exerciseTitle;
```

Comment:
you can simplify this line as follows:
```suggestion
        exerciseTitle = exerciseTitle ?? 'undefined exercise';
```